pkg/deploy/deploy.go

```diff
@@ -188,10 +189,11 @@ func deployComponentsParallel(ctx context.Context, components []graph.SortedComp
 
 	// Iterate over components and launch a goroutine for each component deployment.
 	for i := range components {
-		go func(component graph.SortedComponent) {
+		go func(component graph.SortedComponent, subGraphID int) {
+			ctx = context.WithValue(ctx, log.CtxGraphComponentId{}, log.CtxValGraphComponentId(subGraphID))
```

Comment:
this means we'll only be able to correlate logs if structured logging is active. 
Is there any way for zap to log only specific fields to the console? And in some nicer format than 'key=value' after the message?

It would actually be nice to have the context fields always included in the logs.
Something like this:

```sh
2023-07-25T17:31:45+02:00       info       {envName} {componentID} {coordinate}       Deploying config project:alerting-profile:profile
2023-07-25T17:31:45+02:00       warn       {envName} {componentID} {coordinate}       API for "alerting-profile" is deprecated! Please consider migrating to "builtin:alerting.profile"!
2023-07-25T17:31:45+02:00       debug       {envName} {componentID} {coordinate}       Created/Updated object by ID for Star Trek Service (b0d35095-9065-3ab4-affa-302dce9342b3)
```

Reply:
As discussed we'll do this in a followup PR